prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt

```diff
@@ -127,7 +128,11 @@ class ProcessFhirCommands : CliktCommand(
      * @return a FHIR bundle that represents the data in the one HL7 message
      */
     private fun convertToFhir(hl7String: String, actionLogger: ActionLogger): Bundle {
-        val messages = HL7Reader(actionLogger).getMessages(hl7String)
+        val hasFiveEncodingChars = hl7String.contains("MSH|^~\\&#|")
```

Comment:
This logic makes an assumption that if the batch message contains 1 `MSH|^`\&#`, then all of the messages in the batch have that. In an edge case where we have a batch with a mixture of `MSH|^`\&#` and `MSH|^`\&` and we're wanting to translate one of the `MSH|^`\&` messages, that message will show `MSH|^`\&#` before getting to `translate` and the resulting fhir will mistakenly include a `encoding-characters` extension with value `^`\&#` on it's MessageHeader. This is low impact since it's only in the CLI, and it's likely a valid assumption in most cases. Still in the process of writing this comment, I thought of a different way to do this that might not add too much complexity. Stay tuned for a follow-up commit.

Reply:
Resolved this in my latest commit.